plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/sampler/ProbabilisticSampler.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing.sampler;
+
+import org.opensearch.telemetry.TelemetrySettings;
+
+import java.util.List;
+import java.util.Objects;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.data.LinkData;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+
+/**
+ * ProbabilisticSampler implements a head-based sampling strategy based on provided settings.
+ */
+public class ProbabilisticSampler implements Sampler {
+    private Sampler defaultSampler;
+    private final TelemetrySettings telemetrySettings;
+    private double tracerHeadSamplerSamplingRatio;
+
+    /**
+     * Constructor
+     *
+     * @param telemetrySettings Telemetry settings.
+     */
+    public ProbabilisticSampler(TelemetrySettings telemetrySettings) {
+        Objects.requireNonNull(telemetrySettings);
+        this.telemetrySettings = telemetrySettings;
+        this.tracerHeadSamplerSamplingRatio = telemetrySettings.getTracerHeadSamplerSamplingRatio();
+        this.defaultSampler = createProbabilitySampler(tracerHeadSamplerSamplingRatio);
+    }
+
+    private Sampler createProbabilitySampler(double samplingRatio) {
+        return Sampler.traceIdRatioBased(samplingRatio);
+    }
+
+    Sampler getSampler() {
+        double newSamplingRatio = telemetrySettings.getTracerHeadSamplerSamplingRatio();
+        if (isSamplingRatioChanged(newSamplingRatio)) {
+            synchronized (this) {
+                this.tracerHeadSamplerSamplingRatio = newSamplingRatio;
+                defaultSampler = createProbabilitySampler(tracerHeadSamplerSamplingRatio);
```

Comment:
```suggestion
                defaultSampler = Sampler.traceIdRatioBased(samplingRatio)
```

Reply:
We should not create object of `traceIdRatioBased` sampler. We are calling `createProbabilitySampler` as we have overridden `shouldSample` method of it, to check for updated sampling probability from cluster setting & then sample the data based on new value.

Reply:
Done!